cli/azd/pkg/pipeline/pipeline.go

```diff
@@ -118,6 +120,35 @@ type CiProvider interface {
 	) *CredentialOptions
 }
 
+type ConfigOptions struct {
+	Variables                    []string
+	Secrets                      []string
+	AdditionalVariablesAsSecrets bool
+}
+
+// SecretsAndVar returns the list of variables and secrets to be used in the pipeline
+// The initial values reference azd known values, which are merged with the ones defined on azure.yaml by the user.
+func (c *ConfigOptions) SecretsAndVars(
+	initialVariables, initialSecrets, env map[string]string) (variables map[string]string, secrets map[string]string) {
+	variables = maps.Clone(initialVariables)
+	secrets = maps.Clone(initialSecrets)
+
+	for key, value := range env {
+		if slices.Contains(c.Variables, key) {
```

Comment:
Does it make sense to instead: iterate the variables we need to add, and use the env map to perform the lookup?

Reply:
I think there's no difference.  Probably a sightly performance difference if the env is huge and customer only ask for one value/secret. But, on the other hand, when `AdditionalVariablesAsSecrets`, we would need to iterate the env again at the end to take all non-added entries as secrets.